server/routes/templates.ts

```diff
@@ -555,13 +555,14 @@ function applicationBranchHandler(requestType: ApplicationType, anonymous: boole
 						}
 					}
 				}
+				question["hasResponse"] = savedValue && savedValue.value; // Used to determine whether "Please select" is selected in dropdown lists
```

Comment:
Where does this get used? Are you sure it isn't redundant with 'otherSelected'?

Reply:
I guess what I mean is, is this file used to render the template, does the template regenerate on submission (hence an invalid submission would now have a false 'hasResponse' attribute), and can we really not do this on clientside?

Reply:
`otherSelected` means that the "Other" option at the bottom of the list is selected and for registration to expect a free-form text entry instead. We need `hasResponse` here because the proper information to calculate it within the template isn't exposed to the template. Also, most logic should be in TypeScript and not Handlebars in my opinion